src/libstd/thread.rs

```diff
@@ -414,6 +416,24 @@ impl Thread {
         *guard = false;
     }
 
+    /// Block unless or until the current thread's token is made available or
+    /// the specified duration has been reached (may wake spuriously).
+    ///
+    /// The semantics of this function are equivalent to `park()` except that the
+    /// thread will be blocked for roughly no longer than dur. This method
+    /// should not be used for precise timing due to anomalies such as
+    /// preemption or platform differences that may not cause the maximum
+    /// amount of time waited to be precisely dur
+    ///
+    /// See the module doc for more detail.
+    #[unstable(feature = "std_misc", reason = "recently introduced")]
+    pub fn park_timeout(dur: Duration) {
+        let thread = Thread::current();
+        let mut guard = thread.inner.lock.lock().unwrap();
+        let (mut guard, _) = thread.inner.cvar.wait_timeout(guard, dur).unwrap();
```

Comment:
Shouldn't this be wrapped in `while !*guard` like the loop above? Otherwise this will block no matter what even if a previous `unpark` was seen.